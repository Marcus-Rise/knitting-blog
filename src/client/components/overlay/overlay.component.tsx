import type { FC } from "react";
import React, { useEffect } from "react";
import styles from "./overlay.module.scss";

interface IProps {
  onClose?: () => void;
}

const Overlay: FC<IProps> = ({ children, onClose }) => {
  const onClick = (e: React.MouseEvent): void => {
    e.stopPropagation();

    if (onClose) {
      onClose();
    }
  };

  useEffect(() => {
    const body = document.getElementsByTagName("body")[0];

    body.classList.add(styles.body);

    return () => {
      body.classList.remove(styles.body);
    };
  }, []);

  return (
    <div className={styles.root} onClick={onClick}>
      {children}
    </div>
  );
};

export { Overlay };
