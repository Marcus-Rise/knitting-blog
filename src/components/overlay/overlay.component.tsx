import React, { useEffect } from "react";
import styles from "./overlay.module.scss";

interface IProps {
  onClose?: () => void;
}

const Overlay: React.FC<IProps> = (props) => {
  const onClick = (e: React.MouseEvent): void => {
    e.stopPropagation();

    if (props.onClose) {
      props.onClose();
    }
  };

  useEffect(() => {
    const body = document.getElementsByTagName("body")[0];

    body.classList.add(styles.body);

    return () => {
      body.classList.remove(styles.body);
    };
  }, []);

  return (
    <div className={styles.root} onClick={onClick}>
      {props.children}
    </div>
  );
};

export { Overlay };
