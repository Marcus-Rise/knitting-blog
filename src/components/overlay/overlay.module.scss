.root {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 900;
  width: 100vw;
  height: 100vh;

  background-color: rgba(#58585a, 0.5);
}

.body {
  max-height: 100vh;
  overflow-y: hidden;
}
