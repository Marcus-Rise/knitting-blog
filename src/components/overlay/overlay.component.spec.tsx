import { render } from "@testing-library/react";
import { Overlay } from "./overlay.component";
import React from "react";

describe("Overlay", () => {
  test("render", () => {
    const { asFragment } = render(<Overlay />);

    expect(asFragment()).toMatchSnapshot();
  });
});
