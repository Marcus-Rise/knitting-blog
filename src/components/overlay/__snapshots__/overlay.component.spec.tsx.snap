// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay render 1`] = `
<DocumentFragment>
  <div
    class="root"
  />
</DocumentFragment>
`;
